src/coreclr/gc/gc.cpp

```diff
@@ -41128,7 +41128,8 @@ size_t gc_heap::decommit_region (heap_segment* region, int bucket, int h_number)
     if ((region->flags & heap_segment_flags_ma_committed) != 0)
     {
 #ifdef MULTIPLE_HEAPS
-        gc_heap* hp = heap_segment_heap (region);
+        // For regions, it doesn't matter which heap is used to decommit the mark array
```

Comment:
this is the case with segments too as the mark array is the same for all heaps. I think it would be more useful to say in the comment why you would hit this with regions (because we clear the heap for free regions).